### ISSUE TITLE
Switch from zones to accounts API

### ISSUE THREAD

**Issue by B-Interactive** [B-Interactive/cloudflare-stream-wordpress#11]
I'm opening this for contemplation.  I'm currently not fully convinced a change is needed, but I realise I may also be looking at it wrong.

I've used [the `/zones` API](https://github.com/B-Interactive/cloudflare-stream-wordpress/blob/c1a5914a149ed5704d12e6802cdd0de5569d7002/src/inc/class-cloudflare-stream-api.php#L119) for API calls.  This is as opposed to using the standard `/accounts` API.

Feedback in the [Cloudflare forum](https://community.cloudflare.com/t/upcoming-domain-change-to-ensure-delivery-of-your-video-content/405842/17) prompted me to consider this again.  At the very least, it looks like I need to [update README.md](https://github.com/B-Interactive/cloudflare-stream-wordpress/blob/f1952c247cb6f66b62bb5e44af3f508f65597534/README.md?plain=1#L28), because referencing the `/zones` API as "more secure" doesn't appear to apply to Stream, based on that feedback.

The reason I originally used the `/zones` API, is because it, in conjunction with an API token with _only_ `Account.Stream:Edit` permission, could access some general account data, that is otherwise unavailable using the `/accounts` API.

For example (using `/zones`):
```
curl -X GET "https://api.cloudflare.com/client/v4/zones/{ZONE ID}" -H "Authorization: Bearer {TOKEN}" -H "Content-Type:application/json"
```

Returns data that includes the account ID and account email (among other things).

Using the `/accounts` equivalent on the other hand, produces a permission error, _unless_ additional permissions _beyond_ `Account.Stream:Edit` are granted (I'm not sure which specific one is required):
```
curl -X GET "https://api.cloudflare.com/client/v4/accounts/{ACCOUNT ID}" -H "Authorization: Bearer {TOKEN}" -H "Content-Type:application/json"
```
```
{"success":false,"errors":[{"code":9109,"message":"Unauthorized to access requested resource"}],"messages":[],"result":null}
```

Currently, the `/zones` API is not being used to pull these additional details in this plugin.  In an earlier (non-public) version of my own plugin, I had pulled the account email and ID and simply presented it in the plugin's settings.

However, potentially the same `/zones` API request could be made to grab [the account specific subdomain](https://github.com/B-Interactive/cloudflare-stream-wordpress/issues/7 ).  This is something I might raise with Cloudflare.

**Comment by davidmpurdy** [comment]
I don't think I fully understand how zones are supposed to relate to Stream structurally (if at all). Using zones feels a little strange to me since my understanding is that they're a level down, structurally (Stream is at the account level, zones are at the domain level, right?).

It _is_ easier to get working though (I checked and it seems to work with any of the zone IDs in the account with Stream). I think you're right that's a Cloudflare question - I'd be interested to hear if that's an intentional architecture decision or maybe an accident of implementation.

**Comment by B-Interactive** [comment]
I think you're right @davidmpurdy, it would seem zones aren't supposed to really come into play with Stream.

My implementation using zones was originally at least, taking advantage of what now appears to be an anomaly with that end-point.

Given the official recommendation seems to be to use `/accounts` (for Stream), and currently this plugin does not to take advantage of the data available with `/zones`, **I will commence switching it over to `/accounts`.**

[Cloudflare have said](https://community.cloudflare.com/t/stream-and-accounts-vs-zones-api/414033/2?u=bink) they will look into including an API call that can retrieve [the account specific subdomain](https://github.com/B-Interactive/cloudflare-stream-wordpress/issues/7).

**Comment by B-Interactive** [comment]
I've [pushed a branch with this update](https://github.com/B-Interactive/cloudflare-stream-wordpress/compare/Accounts-API), but I want to run it by you @davidmpurdy first, to make sure this isn't going to impact your current efforts with the blocks.  Please let me know!

If the plugin is already set up with Zone ID and API token, this update will automatically pull and save the Account ID, so no intervention _should_ be needed by the user.  This action is prompted by either of these two things:

1. An admin accesses the plugin's settings page (regardless of whether Save Settings is hit).
2. An API call is made.

In both scenarios, the Account ID is _only_ fetched, if it isn't already in the local database.  It is using `add_option()`, so it will not update an existing Account ID, only an absent one.

**Comment by davidmpurdy** [comment]
That doesn't cause any issues for the block that I can see (this past week got busy and I haven't done much to get an actual PR ready anyway). 

My only question is whether it's worth creating the [base URL](https://github.com/B-Interactive/cloudflare-stream-wordpress/blob/ed38dad2504648f9499bc68b9918824b813f7097/src/inc/class-cloudflare-stream-api.php#L132) in its own function so it wouldn't have to be changed in PHP and JS in the future if it changed again? Maybe even including the account ID so the two places we interface with the API just need to be aware of endpoint/authentication/payload? Or maybe that's over-thinking it since it's not likely to change often if at all.

**Comment by B-Interactive** [comment]
Thanks @davidmpurdy , that's a good point regarding the base URL, and it would actually benefit solutions for the [variable sub-domains](https://github.com/B-Interactive/cloudflare-stream-wordpress/issues/7), or even Cloudflare URL options (videodelivery.net, cloudflarestream.com) which may be a part of that solution too.

I'll take another look at it and see about eliminating redundancy on the current doubling up of those request functions too.

**Comment by B-Interactive** [comment]
Completed with 8aec255043ced8b65abf17c7a2889eaef37ca1a2.

**Comment by davidmpurdy** [comment]
Minor bug with trying to upgrade to the account ID when it doesn't exist. 

Fixed with #13

**Comment by B-Interactive** [comment]
Thanks for picking that up, and providing the patch!
